### PR TITLE
Feature/Implement http client

### PR DESCRIPTION
**Pull request by Miroshni4enko**
- [x] Add tests
- [x] Delete `Main` class
- [x] Add `Http` class
- [x] Check with linters

### CODE REVIEW

File: src/main/java/org/truemail/Client.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        result.append(this.configuration.getPort());
+        result.append("?email=");
+        result.append(email);
+
+        return result.toString();
+    }
+
+    private Map<String, String> headers() {
+        return new HashMap<String, String>() {
+            {
```
Comment (bestwebua) [inline src/main/java/org/truemail/Client.java:63]: Double-brace initialization should be avoided.

File: src/test/java/org/truemail/ClientTests.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        assertEquals("Be Nice", response);
+    }
+
+    @Test
+    public void validateReturnsErrorResponseWhenHttpConnectionFailed() throws Exception {
+        URL u = mock(URL.class);
+        String url = "https://test.host:12345?email=test.email";
+        whenNew(URL.class).withArguments(url).thenReturn(u);
+        HttpURLConnection huc = mock(HttpURLConnection.class);
```
Comment (bestwebua) [inline src/test/java/org/truemail/ClientTests.java:41]: Needless, in this case you don't huc var.

File: src/test/java/org/truemail/ClientTests.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.truemail.client.Configuration;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static junit.framework.Assert.assertEquals;
+import static org.powermock.api.mockito.PowerMockito.*;
```
Comment (bestwebua) [inline src/test/java/org/truemail/ClientTests.java:14]: Please avoid unused import.

File: src/main/java/org/truemail/client/Configuration.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+public class Configuration {
+    public static int DEFAULT_PORT = 9292;
+
+    private boolean secureConnection;
+    private String host;
+    private int port;
+    private String token;
+
+    public Configuration(boolean secureConnection, String host, int port, String token) {
```
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply src/main/java/org/truemail/client/Configuration.java:11]: It used in the next constructor

[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply src/main/java/org/truemail/client/Configuration.java:11]: Java does not have a default argument(
Maybe the new one has

File: src/main/java/org/truemail/Client.java [hunk 5]
```diff
@@ -1,46 +1,26 @@
...
 import javax.net.ssl.HttpsURLConnection;
 import java.io.IOException;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.util.HashMap;
 import java.util.Map;
 
-public class Http {
+public class Client {
     public static String USER_AGENT = "Truemail Crystal client";
```
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply src/main/java/org/truemail/Client.java:14]: Good catch, thanks!)

File: src/test/java/org/truemail/ClientTests.java (hunk 2 above)
Comment (Miroshni4enko) [reply src/test/java/org/truemail/ClientTests.java:41]: Good catch, interesting why IDE does not highlight this for me....

File: src/main/java/org/truemail/Client.java (hunk 1 above)
Comment (Miroshni4enko) [reply src/main/java/org/truemail/Client.java:63]: Interesting, I will refactor it

File: src/test/java/org/truemail/ClientTests.java (hunk 2 above)
Comment (Miroshni4enko) [reply src/test/java/org/truemail/ClientTests.java:41]: Good catch, thanks!)

File: src/test/java/org/truemail/ClientTests.java (hunk 3 above)
Comment (Miroshni4enko) [reply src/test/java/org/truemail/ClientTests.java:14]: Actually, it used for methods `when` and `mock`

**Review by bestwebua — APPROVED** [review]

**Review by bestwebua — APPROVED** [review]

**Pull request merged**